include/pybind11/docstring_options.h

```diff
@@ -0,0 +1,73 @@
+/*
+    pybind11/docstring_options.h: selectively suppress the automatic generation of docstrings and function signatures.
+
+    Copyright (c) 2016 Wenzel Jakob <wenzel.jakob@epfl.ch>
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#pragma once
+
+#include "common.h"
+
+NAMESPACE_BEGIN(pybind11)
+
+class docstring_options
+{
+  public:
+
+    struct state {
+        bool show_user_defined; //< Include user-defined texts in docstrings.
+        bool show_signatures;   //< Inclcude auto-generated function signatures in docstrings.
+    };
+
+    PYBIND11_NOINLINE inline static state &global_state() {
```

Comment:
`PYBIND11_NOINLINE inline` isn't needed here.

Although perhaps you could just use a `static state global_state = { true, true };`, i.e. not protected behind a method?